riscv/src/continuations.rs

```diff
@@ -235,6 +235,12 @@ pub fn rust_continuations_dry_run<F: FieldElement>(
         .fold(None, |acc, (_, m)| acc.or(m.degree.clone()))
         .unwrap()
         .degree;
+
+    let length = match length {
+        Expression::Number(length, None) => length,
+        _ => unimplemented!(),
+    };
```

Comment:
@leonardoalt @georgwiese do you have a suggestion here? This would break if the degree is not a literal

Reply:
is this already resolved? Say for like `2**18`? I think this will be the main use

Reply:
Everything works now, but this would break if for some reason the degree is more complex as it doesn't get evaluated

Reply:
Sorry for the late reply!

At what point does it get evaluated? We could get that artifact from the pipeline instead, e.g. the analyzed PIL?

It seems a bit outrageous to do the whole conversion to PIL just to get that number. On the other hand, in practice we will run `rust_continuations()` next anyway, which also needs that artifact.

Reply:
@georgwiese it is evaluated [here](https://github.com/powdr-labs/powdr/blob/d2bd95cfd5d3e67c2585e55e73742e94a30f354f/pil-analyzer/src/pil_analyzer.rs#L311) and there is already a TODO to evaluate it earlier but I think it should be a different PR.